backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/database/jpa/PostgresDocumentationUnitRepositoryImpl.java

```diff
@@ -681,11 +536,391 @@ public List<UUID> getRandomDocumentationUnitIds() {
+    if (decisionDateEnd != null) {
+      decisionDatePredicate =
+          criteriaBuilder.between(root.get("date"), decisionDate, decisionDateEnd);
+    }
+    conditions = criteriaBuilder.and(conditions, decisionDatePredicate);
+    return conditions;
+  }
+
+  private static Predicate addDocOfficeFilter(
+      CriteriaBuilder criteriaBuilder,
+      Root<DocumentationUnitDTO> root,
+      UUID docOfficeId,
+      Predicate conditions) {
+    Predicate documentOfficePredicate =
+        criteriaBuilder.equal(root.get(DOCUMENTATION_OFFICE).get("id"), docOfficeId);
+    conditions = criteriaBuilder.and(conditions, documentOfficePredicate);
+    return conditions;
+  }
+
+  private static Predicate addScheduledFilter(
+      CriteriaBuilder criteriaBuilder, Root<DocumentationUnitDTO> root, Predicate conditions) {
+    Predicate scheduledPredicate =
+        criteriaBuilder.isNotNull(root.get(SCHEDULED_PUBLICATION_DATE_TIME));
+    conditions = criteriaBuilder.and(conditions, scheduledPredicate);
+    return conditions;
+  }
+
+  private static Predicate addWithErrorFilter(
+      CriteriaBuilder criteriaBuilder,
+      Root<DocumentationUnitDTO> root,
+      UUID documentationOfficeId,
+      Predicate conditions) {
+    Join<DocumentationUnitDTO, String> statusJoin = root.join(STATUS, JoinType.LEFT);
+    Predicate errorPredicate =
+        criteriaBuilder.and(
+            criteriaBuilder.equal(root.get(DOCUMENTATION_OFFICE).get("id"), documentationOfficeId),
+            criteriaBuilder.isTrue(statusJoin.get("withError")));
+    conditions = criteriaBuilder.and(conditions, errorPredicate);
+    return conditions;
+  }
+
+  private static Predicate addDuplicateFilter(
+      CriteriaBuilder criteriaBuilder,
+      Root<DocumentationUnitDTO> root,
+      UUID documentationOfficeId,
+      Predicate conditions) {
+    Join<DocumentationUnitDTO, DuplicateRelationDTO> duplicateRelation1 =
+        root.join("duplicateRelations1", JoinType.LEFT);
+    Join<DocumentationUnitDTO, DuplicateRelationDTO> duplicateRelation2 =
+        root.join("duplicateRelations2", JoinType.LEFT);
+    Predicate errorPredicate =
+        criteriaBuilder.and(
+            criteriaBuilder.equal(root.get(DOCUMENTATION_OFFICE).get("id"), documentationOfficeId),
+            criteriaBuilder.or(
+                criteriaBuilder.equal(
+                    duplicateRelation1.get(STATUS),
+                    criteriaBuilder.literal(DuplicateRelationStatus.PENDING)),
+                criteriaBuilder.equal(
+                    duplicateRelation2.get(STATUS),
+                    criteriaBuilder.literal(DuplicateRelationStatus.PENDING))));
+    conditions = criteriaBuilder.and(conditions, errorPredicate);
+    return conditions;
+  }
+
+  private static Predicate addPublicationDateFilter(
+      CriteriaBuilder criteriaBuilder,
+      Root<DocumentationUnitDTO> root,
+      LocalDate publicationDate,
+      Predicate conditions) {
+    Predicate publicationDatePredicate =
```

Comment:
Here is the other part to withError. The publication date only allowed to search if you activate scheduledOnly, right?